File: src/main/java/com/nbteam/killer/helper/domain/GameBook.java
package com.nbteam.killer.helper.domain;

import cn.hutool.core.util.StrUtil;
import com.nbteam.killer.helper.enums.RoleEnum;
import lombok.Data;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

@Data
public class GameBook implements Serializable {

    String id;

    String title;

    public GameBook(RoleEnum... rs) {
        Arrays.stream(rs).forEach(r -> players.add(new Player(r)));
    }

    public GameBook(String title, RoleEnum... rs) {
        this.title = title;
        Arrays.stream(rs).forEach(r -> players.add(new Player(r)));
    }

    public String getId() {

        if (StrUtil.isBlankOrUndefined(id)) {
            // id = UUID.randomUUID().toString().replace("-", "").toLowerCase();
            id = String.valueOf(title.hashCode());
        }

        return id;
    }

    List<Player> players = new ArrayList<>();

    public Integer getTotal() {
        return players.size();
    }

}

File: killer-helper-base/src/main/kotlin/com/nbteam/killer/helper/base/domain/GameBook.kt
package com.nbteam.killer.helper.base.domain

import cn.hutool.core.util.StrUtil
import com.nbteam.killer.helper.base.enums.RoleEnum
import lombok.Data
import java.io.Serializable
import java.util.*

@Data
open class GameBook : Serializable {
    private var id: String? = null
    var title: String? = null

    constructor(vararg rs: RoleEnum) {
        Arrays.stream(rs).forEach { r: RoleEnum -> players.add(Player(r)) }
    }

    constructor(title: String?, vararg rs: RoleEnum) {
        this.title = title
        Arrays.stream(rs).forEach { r: RoleEnum -> players.add(Player(r)) }
    }

    fun getId(): String? {
        if (StrUtil.isBlankOrUndefined(id)) {
            id = UUID.randomUUID().toString().replace("-", "").lowercase(Locale.getDefault())
        }
        return id
    }

    open var players: MutableList<Player> = ArrayList<Player>()
    val total: Int
        get() = players.size
}